[t628] initial dom fragment work

secretrobotron opened the pull request:
1. Changed media-view.js to use DOMFragment
2. DOMFragment module
3. text require.js plugin
4. layouts folder, with media-view.xml

secretrobotron committed:
[t628] initial dom fragment work

1. Changed media-view.js to use DOMFragment
2. DOMFragment module
3. text require.js plugin
4. layouts folder, with media-view.xml

secretrobotron committed:
[t628] lint fixes

jbuck commented on make.js:
```diff
@@ -36,6 +36,9 @@ target['check-lint'] = function() {
   echo('### Linting JS files');
 
   var files = find('src').filter( function( file ) {
+    if(file === "src/text.js"){
```
Move the require.js text plugin out of src/text.js so that we don't need to include this filter in the makefile

jbuck commented on src/core/views/media-view.js:
```diff
@@ -1,50 +1,25 @@
-define( [ "ui/page-element", "ui/logo-spinner" ], function( PageElement, LogoSpinner ){
-  
+define( [ "ui/page-element", "ui/logo-spinner", "util/dom-fragment", "text!layouts/media-view.xml" ],
```
Change the extension to .html instead of .xml

jbuck commented on src/util/dom-fragment.js:
```diff
@@ -0,0 +1,20 @@
+
+  return function( inputString ) {
+    var renderContainer = document.createElement( "div" );
+    renderContainer.innerHTML = inputString;
```
Use createContextualFragment instead; then you don't need to worry about removing this wrapper div

secretrobotron committed:
[t628] review changes

1. Moved require text plugin.
2. Renamed media-view to be an html file
3. Moved Range functionality to util/lang

secretrobotron committed:
[t628] removed special-case for text plugin

secretrobotron committed:
[t628] lint fix

secretrobotron committed:
[t628] moved require path and context to config.js

secretrobotron committed:
[t628] fixing require text problems

1. New require.js
2. Trimmed up config.js
3. Fixed domFragment code to return severed firstChild
4. Updated media-view.js (was broken)

secretrobotron committed:
[t628] fixed build system for text plugin
